### PR TITLE
Add py-atlannot

### PR DESCRIPTION
**Pull request by Stannislav**
Repository: https://bbpgitlab.epfl.ch/project/proj101/atlas_annotation

### CODE REVIEW

**Comment by Stannislav** [comment]
I'm not sure about the build failure and what to do with it.

The failing stage is "Post Mortem / Test Modules", and the short error log I can see is:
```
Building hpc.module-testing

[2021-09-15T16:13:59.995Z] Scheduling project: hpc.module-testing
[2021-09-15T16:14:09.278Z] Starting building: hpc.module-testing #4038
hpc.module-testing #4038 completed with status FAILURE (propagate: false to ignore)
```
Maybe there's something wrong with my `applications.yaml`?

File: var/spack/repos/builtin/packages/py-atlannot/package.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    version('0.1.1', commit='5271681de15f23e1df6f8556405b03218541d241')
+
+    depends_on('python@3.7:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-setuptools-scm', type='build')
+
+    # Installation requirements
+    depends_on('py-antspyx@0.2.4', type=('run'))
+    depends_on('py-atldld@0.2.2', type=('run'))
+    # depends_on('py-dvc', type=('run'))
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-atlannot/package.py:27]: What's up with this?

**Comment by matz-e** [comment]
> I'm not sure about the build failure and what to do with it.
> 
> The failing stage is "Post Mortem / Test Modules", and the short error log I can see is:
> 
> ```
> Building hpc.module-testing
> 
> [2021-09-15T16:13:59.995Z] Scheduling project: hpc.module-testing
> [2021-09-15T16:14:09.278Z] Starting building: hpc.module-testing #4038
> hpc.module-testing #4038 completed with status FAILURE (propagate: false to ignore)
> ```
> 
> Maybe there's something wrong with my `applications.yaml`?

You need to look here: https://bbpcode.epfl.ch/ci/blue/organizations/jenkins/hpc.module-testing/detail/hpc.module-testing/4038/pipeline

This is a timeout that happens quite often :(

Retest this please

File: var/spack/repos/builtin/packages/py-atlannot/package.py (hunk 1 above)
Comment (Stannislav) [reply var/spack/repos/builtin/packages/py-atlannot/package.py:27]: Ah yes, it's a package that's in theory listed in the requirements in `setup.py`, but it's not really needed to use the library, only to manually fetch the input data.

So, I could do one of these things:
- Delete this line
- Deploy `py-dvc`, but like I said, we don't really need this now
- Remove this dependency from the install requirements of the library and maybe put it inside `extra_requires`.

Comment (matz-e) [reply var/spack/repos/builtin/packages/py-atlannot/package.py:27]: If it's in `setup.py`, then our distribution check should have complained. A comment may be nice, but for now I'm fine with it as it is.

**Review by matz-e — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-atlannot/package.py (hunk 1 above)
Comment (Stannislav) [reply var/spack/repos/builtin/packages/py-atlannot/package.py:27]: Yes, if it's OK, let's leave it there as a reminder. I'll also create a ticket on our JIRA to do something about this requirement in the future.

**Pull request merged**